src/main/java/macrobase/analysis/stats/distribution/MultivariateTDistribution.java

```diff
@@ -0,0 +1,60 @@
+package macrobase.analysis.stats.distribution;
+
+import org.apache.commons.math3.linear.LUDecomposition;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MultivariateTDistribution {
+    private static final Logger log = LoggerFactory.getLogger(MultivariateTDistribution.class);
+    private RealVector mean;
+    private RealMatrix covarianceMatrix;
+    private RealMatrix precisionMatrix;
+    private double degreesOfFreedom;
+
+    private int dimensions;
+    private double multiplier;
+
+    public MultivariateTDistribution(RealVector mean, RealMatrix covarianceMatrix, int degreesOfFreedom) {
+        this.mean = mean;
+        this.covarianceMatrix = covarianceMatrix;
+        if (mean.getDimension() > 1) {
+            this.precisionMatrix = MatrixUtils.blockInverse(covarianceMatrix, (-1 + covarianceMatrix.getColumnDimension()) / 2);
+        } else {
+            this.precisionMatrix = MatrixUtils.createRealIdentityMatrix(1).scalarMultiply(1. / covarianceMatrix.getEntry(0, 0));
+        }
+        this.degreesOfFreedom = degreesOfFreedom;
+
+        this.dimensions = mean.getDimension();
+
+        double determinant = new LUDecomposition(covarianceMatrix).getDeterminant();
+
+        this.multiplier = halfGammaRatio(dimensions + degreesOfFreedom, degreesOfFreedom)/
+                Math.pow(Math.PI * degreesOfFreedom, 0.5 * dimensions) /
+                Math.pow(determinant, 0.5);
+    }
+
+
+    public double density(RealVector vector) {
```

Comment:
They have too many methods in the abstract class from apache, so let's keep it simple for now.